tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -91,8 +91,13 @@ private static boolean shouldObserve(Observability observability) {
         throw new SafeIllegalArgumentException("Unknown observability", SafeArg.of("observability", observability));
     }
 
-    @Beta
-    static TraceMetadata getTraceMetadata() {
+    /**
+     * In the unsampled case, the Trace.Unsampled class doesn't actually store a spanId/parentSpanId
+     * stack, so we just make one up (just in time). This matches the behaviour of Tracer#startSpan.
+     * <p>
+     * n.b. this is a bit funky because calling getTraceMetadata multiple times will return different spanIds
+     */
```

Comment:
Let's document that callers must have prior knowledge that there's an active trace with a root span, otherwise this throws.

Should we return Optional<TraceMetadata> instead of throwing to avoid misuse?